tests/unit_tests/test_wave_06.py

```diff
@@ -228,7 +237,18 @@ def test_swap_best_by_category_no_match_is_false():
         their_priority="Clothing"
     )
 
-    raise Exception("Complete this test according to comments below.")
+    tai_list = [item_a, item_b, item_c]
+    jesse_list = [item_d, item_e, item_f]
+
+    assert not result
+    assert len(tai.inventory) == 3
+    assert len(jesse.inventory) == 3
+    assert all (item in tai.inventory for item in tai_list)
+    assert all (item in jesse.inventory for item in jesse_list)
```

Comment:
👀 Using `all` like this is a nice way to concisely check that all the listed items are present, which means that it doesn't even matter whether `.inventory` is a list, just that each iterated item is `in` it.

However, for this project we still request that folks avoid extra functions like this.

The approach is really nice, so if you'd like to keep it, you could implement your own version of `all`, maybe `my_all` (you know enough to be able to write it!). Or you could switch to an approach that doesn't need `all`.

Also, notice that the expression providing the input to `all` is not a list comprehension (which would produce a list). It's a generator. So you should also confirm that you are clear on how a list comprehension and a generator differ.

Reply:
Also, `all` is a function, so it would be written as `all()` (the parens abutting the name) rather than with a space.